sound/soc/codecs/rt715-sdca.c

```diff
@@ -223,33 +223,24 @@ static int rt715_sdca_set_amp_gain_get(struct snd_kcontrol *kcontrol,
 static const DECLARE_TLV_DB_SCALE(in_vol_tlv, -17625, 375, 0);
 static const DECLARE_TLV_DB_SCALE(mic_vol_tlv, 0, 1000, 0);
 
-static int rt715_sdca_fu_info(struct snd_kcontrol *kcontrol,
-			 struct snd_ctl_elem_info *uinfo)
-{
-	uinfo->type = SNDRV_CTL_ELEM_TYPE_INTEGER;
-	uinfo->count = 2;
-	uinfo->value.integer.min = 0;
-	uinfo->value.integer.max = 1;
-
-	return 0;
-}
-
 static int rt715_sdca_get_volsw(struct snd_kcontrol *kcontrol,
 	struct snd_ctl_elem_value *ucontrol)
 {
 	struct snd_soc_component *component = snd_kcontrol_chip(kcontrol);
 	struct soc_mixer_control *mc =
 		(struct soc_mixer_control *)kcontrol->private_value;
+	unsigned int reg = mc->reg;
+	unsigned int reg2 = mc->rreg;
 	unsigned int invert = mc->invert;
-	unsigned int max = mc->max;
+	int max = mc->max;
 	int val;
 
-	val = snd_soc_component_read(component, mc->reg);
+	val = snd_soc_component_read(component, reg);
 	if (val < 0)
 		return -EINVAL;
```

Comment:
don't know if this fixup would be a good place for this, but wouldn't it be better to propagate `ret` here?